enterprise/internal/batches/webhooks/changeset.go

```diff
@@ -36,64 +35,144 @@ type changeset struct {
+	}
+	Errors []gqlerrors.FormattedError
+}
+
+func marshalChangeset(ctx context.Context, client httpcli.Doer, id graphql.ID) ([]byte, error) {
+	q := queryInfo{
+		Name:      "Changeset",
+		Query:     gqlChangesetQuery,
+		Variables: map[string]any{"id": id},
 	}
 
-	var (
-		title       *string
-		body        *string
-		authorName  *string
-		externalURL *string
-	)
+	var res gqlChangesetResponse
+	if err := makeRequest(ctx, q, client, &res); err != nil {
+		return nil, err
+	}
 
-	if cs.Published() && !cs.IsImporting() {
-		for name, field := range map[string]struct {
-			method func() (string, error)
-			out    **string
-		}{
-			"title":        {cs.Title, &title},
-			"body":         {cs.Body, &body},
-			"author name":  {cs.AuthorName, &authorName},
-			"external URL": {cs.URL, &externalURL},
-		} {
-			value, err := field.method()
-			if err != nil {
-				return nil, errors.Wrapf(err, "getting %s", name)
-			}
-			*field.out = &value
+	if len(res.Errors) > 0 {
+		var combined error
+		for _, err := range res.Errors {
+			combined = errors.Append(combined, err)
 		}
+		return nil, combined
 	}
 
-	payload := changeset{
-		ID:                  bgql.MarshalChangesetID(cs.ID),
-		ExternalID:          cs.ExternalID,
+	node := res.Data.Node
+
+	var batchChangeIDs []graphql.ID
+	for _, bc := range node.BatchChanges.Nodes {
+		batchChangeIDs = append(batchChangeIDs, bc.ID)
+	}
+
+	var labels []string
+	for _, label := range node.Labels {
+		labels = append(labels, label.Text)
+	}
+
+	var authorName *string
+	if node.Author != nil {
+		authorName = &node.Author.Name
+	}
+
+	var externalURL *string
+	if node.ExternalURL != nil {
+		externalURL = &node.ExternalURL.URL
+	}
+
+	return json.Marshal(changeset{
+		ID:                  node.ID,
+		ExternalID:          node.ExternalID,
 		BatchChangeIDs:      batchChangeIDs,
-		OwningBatchChangeID: nullableMap(cs.OwnedByBatchChangeID, bgql.MarshalBatchChangeID),
-		RepositoryID:        bgql.MarshalRepoID(cs.RepoID),
-		CreatedAt:           cs.CreatedAt,
-		UpdatedAt:           cs.UpdatedAt,
-		Title:               title,
-		Body:                body,
+		RepositoryID:        node.Repository.ID,
+		CreatedAt:           node.CreatedAt,
+		UpdatedAt:           node.UpdatedAt,
+		Title:               node.Title,
+		Body:                node.Body,
 		AuthorName:          authorName,
-		State:               string(cs.State),
-		Labels:              labelNames,
+		State:               node.State,
+		Labels:              labels,
 		ExternalURL:         externalURL,
-		ForkNamespace:       nullable(cs.ExternalForkNamespace),
-		ForkName:            nullable(cs.ExternalForkName),
-		ReviewState:         nullable(string(cs.ExternalReviewState)),
-		CheckState:          nullable(string(cs.ExternalCheckState)),
-		Error:               cs.FailureMessage,
-		SyncerError:         cs.SyncErrorMessage,
-	}
-
-	return json.Marshal(&payload)
+		ForkName:            node.ForkName,
+		OwningBatchChangeID: node.OwnedByBatchChange,
```

Comment:
I just noticed, shouldn't these field names match?

```suggestion
		OwnedByBatchChange: node.OwnedByBatchChange,
```